### PR TITLE
Add numProblems slider

### PR DESCRIPTION
**Pull request by jasonz6688**
### List of Changes:

- Added slider to the lobby page
- Added numProblems value to the room, lobby, and updateSettingsRequest
- Added two tests, one that tests the correct number of problems is returned, one that tests that an error is thrown when numProblems is too large or too small

### Notes:

The slider is set between 1 and 10, as that's what the pre-written code in the backend accepted. I kept all the backend code.

### Screencast and Images:

Cap63
Cap64

### CODE REVIEW

File: frontend/src/api/Game.ts
Comment (alankbi) [inline frontend/src/api/Game.ts:118]: nit: re-add new line

File: frontend/src/views/Lobby.tsx
Comment (alankbi) [inline frontend/src/views/Lobby.tsx:146]: nit: this can be simply of type `<number>` (or no type at all, since it can infer that it's a number from the default value)

File: src/test/java/com/rocketden/main/service/GameManagementServiceTests.java
Comment (alankbi) [inline src/test/java/com/rocketden/main/service/GameManagementServiceTests.java:208]: Could you check if there are any existing tests that cover this? My intuition would be that we've already written tests to cover setting numProblems and invalid numProblem errors - however it's been a while so I could be wrong on this, in which case thanks for adding these two tests :)

**Review by alankbi — APPROVED** [review]
Looks good, approved! However, I left a few nits which you should take a look at before merging (also would recommend waiting till @zpChris reviews as well in case he has any other comments)

**Review by zpChris — APPROVED** [review]
LGTM! ✅ Just left two small notes, definitely respond to @alankbi's comments before merging 👍

File: src/test/java/com/rocketden/main/service/GameManagementServiceTests.java
Comment (jasonz6688) [reply src/test/java/com/rocketden/main/service/GameManagementServiceTests.java:208]: So the most similar tests I found were a couple of get problem tests in ProblemServiceTests, which tested whether multiple problems are sent from the backend. Also I could theoretically just modify the startGameSuccess test to include multiple questions. What do you think I should do about this?

File: frontend/src/views/Lobby.tsx
Comment (jasonz6688) [reply frontend/src/views/Lobby.tsx:764]: Good point!

File: frontend/src/views/Lobby.tsx
Comment (jasonz6688) [reply frontend/src/views/Lobby.tsx:769]: I added this because all our other sliders seem to have another check, despite the number still being restricted by the slider and unallowable answers caught by the backend. I could delete this if statement, but I don't think there is much harm in leaving it in here.

Comment (zpChris) [reply frontend/src/views/Lobby.tsx]: Got it - agreed, that makes sense, leave it in! 👍

File: src/test/java/com/rocketden/main/service/GameManagementServiceTests.java
Comment (alankbi) [reply src/test/java/com/rocketden/main/service/GameManagementServiceTests.java:208]: Nah that's fine, your explanation makes sense!

**Pull request merged**
